setup.py

```diff
@@ -332,20 +326,14 @@ def finalize_options(self):
         if self.debug:
             global DEBUG
             DEBUG = True
-        if sys.version_info.major >= 3 and not self.parallel:
-            # For Python 2.7, we monkeypatch distutils to have parallel
-            # builds. If --parallel (or -j) wasn't specified, we want to
-            # reproduce the same behavior as before, that is, auto-detect the
-            # number of jobs.
-            self.parallel = mp_compile.MAX_PROCS
```

Comment:
Is this actually unused? `mp_compile.MAX_PROCS` is defined as `MAX_PROCS = int(os.environ.get("MAX_CONCURRENCY", min(4, cpu_count())))`, and `self.parallel` is from distutils if a parameter was passed on the command line.

Looking at AppVeyor, it looks like the build is now serial instead of parallel, and about 10% slower in total.

Reply:
I'll double check. Is there something in the logs that shows serial vs. parallel.

Reply:
At the start of build_ext, it says 'building <...> extension'. If you see only one such line, it is serial. In #4084 I see it twice, so it is using two threads.

Reply:
Only checking 3.7 on AppVeyor, I see the same thing for this #4084, this PR and master:

```console
$ grep "building .* extension" log-*
log-4084.txt:[00:05:37] building 'PIL._imaging' extension
log-4084.txt:[00:05:37] building 'PIL._imagingft' extension
log-4084.txt:[00:05:37] building 'PIL._webp' extension
log-4084.txt:[00:05:37] building 'PIL._imagingtk' extension
log-4084.txt:[00:05:37] building 'PIL._imagingmath' extension
log-4084.txt:[00:05:37] building 'PIL._imagingmorph' extension

log-4109.txt:[00:05:37] building 'PIL._imaging' extension
log-4109.txt:[00:05:37] building 'PIL._imagingft' extension
log-4109.txt:[00:05:37] building 'PIL._webp' extension
log-4109.txt:[00:05:37] building 'PIL._imagingtk' extension
log-4109.txt:[00:05:37] building 'PIL._imagingmath' extension
log-4109.txt:[00:05:37] building 'PIL._imagingmorph' extension

log-master.txt:[00:04:54] building 'PIL._imaging' extension
log-master.txt:[00:04:54] building 'PIL._imagingft' extension
log-master.txt:[00:04:54] building 'PIL._webp' extension
log-master.txt:[00:04:54] building 'PIL._imagingtk' extension
log-master.txt:[00:04:54] building 'PIL._imagingmath' extension
log-master.txt:[00:04:54] building 'PIL._imagingmorph' extension
```

Reply:
I should have clarified, the timing is what is important here. On master and on #4084, there are two `building <...> extension` lines right at the start, one for each thread, whereas this PR only has one such line at the start. Afterwards you see the compile commands for the extensions being built. The other `building <...> extension` lines come in after each extension is done building. You could check this with timestamps, if they weren't useless on AppVeyor (`winbuild/build.py` collects the entire build_ext output and dumps it all at once).

master: https://ci.appveyor.com/project/Python-pillow/pillow/builds/27820023/job/9t8kptc3rhrsq3ox?fullLog=true#L5797
this PR: https://ci.appveyor.com/project/Python-pillow/pillow/builds/27822685/job/uogulucbfa30x2ka?fullLog=true#L5793

Reply:
Reinstated and simplified in dcfb65f1cb3f0203e8139b603195a1ad9170f9bf.

Replace `multiprocessing.cpu_count()` with `os.cpu_count()` so we can skip `import multiprocessing`, and the former is a wrapper around the latter:

https://github.com/python/cpython/blob/4452cdcb79fdade13c00eabe63304efa4da1c3ad/Lib/multiprocessing/context.py#L39-L45

When `os.cpu_count()` returns `None`, `multiprocessing.cpu_count()` raises `NotImplementedError`.

And `min(4, None)` and `int(None)` raise `TypeError`, so switch our catcher.